cmd/vms.go

```diff
@@ -55,12 +68,54 @@ func (c VMsCmd) printDeployments(instTable InstanceTable) error {
 	return nil
 }
 
-func (c VMsCmd) printDeployment(dep boshdir.Deployment, instTable InstanceTable) error {
-	vmInfos, err := dep.VMInfos()
+type deploymentInfo struct {
+	depName string
+	vmInfos []boshdir.VMInfo
+}
+
+func parallelVMInfos(deployments []boshdir.Deployment, parallel int) (map[string][]boshdir.VMInfo, error) {
+	if parallel == 0 {
+		parallel = 5
+	}
+	workSize := len(deployments)
+	resultc := make(chan deploymentInfo, workSize)
+	errorc := make(chan error, workSize)
+	defer close(resultc)
+	defer close(errorc)
+	works := make([]func(), workSize)
+
+	for i, dep := range deployments {
+		dep := dep
+		works[i] = func() {
+			vmInfos, err := dep.VMInfos()
+			errorc <- err
+			resultc <- deploymentInfo{dep.Name(), vmInfos}
+		}
+	}
+
+	throttler, err := workpool.NewThrottler(parallel, works)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	throttler.Work()
+	vms := make(map[string][]boshdir.VMInfo, workSize)
+	var vmInfoErrors []string
+	for i := 0; i < workSize; i++ {
+		errc := <-errorc
+		result := <-resultc
+		if errc != nil {
+			vmInfoErrors = append(vmInfoErrors, errc.Error())
+		}
+		vms[result.depName] = result.vmInfos
 	}
+	if len(vmInfoErrors) > 0 {
+		err := fmt.Errorf("%s", strings.Join(vmInfoErrors, "\n"))
```

Comment:
This seems like a good candidate for [`bosherr.MultiError`](https://github.com/cloudfoundry/bosh-utils/blob/master/errors/multi_error.go) - what do you think?